app/src/androidTest/java/com/github/houseorganizer/houseorganizer/EditHouseholdTest.java

```diff
@@ -0,0 +1,389 @@
+import static androidx.test.espresso.intent.Intents.intended;
+import static androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent;
+import static androidx.test.espresso.matcher.ViewMatchers.isClickable;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static com.github.houseorganizer.houseorganizer.FirebaseTestsHelper.TEST_HOUSEHOLD_NAMES;
+import static com.github.houseorganizer.houseorganizer.FirebaseTestsHelper.TEST_USERS_EMAILS;
+import static com.github.houseorganizer.houseorganizer.FirebaseTestsHelper.UNKNOWN_USER;
+import static com.github.houseorganizer.houseorganizer.FirebaseTestsHelper.WRONG_EMAIL;
+import com.github.houseorganizer.houseorganizer.house.HouseSelectionActivity;
+import com.github.houseorganizer.houseorganizer.house.EditHousehold;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import android.content.Intent;
+
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.espresso.intent.Intents;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+@RunWith(AndroidJUnit4.class)
+public class EditHouseholdTest {
+    private static FirebaseFirestore db;
+    private static FirebaseAuth auth;
+    private static Intent intentFromHouseSelection;
+
+    @BeforeClass
+    public static void createMockFirebase() throws ExecutionException, InterruptedException {
+        FirebaseTestsHelper.startAuthEmulator();
+        FirebaseTestsHelper.startFirestoreEmulator();
+        FirebaseTestsHelper.setUpFirebase();
+
+        db = FirebaseFirestore.getInstance();
+        auth = FirebaseAuth.getInstance();
+        intentFromHouseSelection = new Intent(ApplicationProvider.getApplicationContext(), EditHousehold.class).putExtra(HouseSelectionActivity.HOUSEHOLD_TO_EDIT, TEST_HOUSEHOLD_NAMES[0]);
+    }
+
+    @AfterClass
+    public static void signOut(){
+        auth.signOut();
+    }
+
+    @Rule
+    public ActivityScenarioRule<EditHousehold> editHouseholdRule = new ActivityScenarioRule<>(intentFromHouseSelection);
+
+    @Before
+    public void setupHouseholds() throws ExecutionException, InterruptedException {
+        FirebaseTestsHelper.createHouseholds();
+    }
+
+    @After
+    public void dismantleHouseholds() throws ExecutionException, InterruptedException {
+        FirebaseTestsHelper.createHouseholds();
+    }
+
+    @Test
+    public void addUserIsDisplayed() {
```

Comment:
Do the display tests work all the time? I remember we had some problems with buttons on main screen failing this kind of test. Also, another question: what do you think of "squishing" displayed / clickable tests into one function? This is what I'm thinking of doing when testing the UI for task lists.

Reply:
Yes, the tests do work every time they are run I don’t need to use thread sleep for some reason

Reply:
I was just curious if you'd be open to it. In any case, I don't think there's a need to make these changes right now.